### PR TITLE
Hotfixes for DOM strategy and code refactoring

### PR DESCRIPTION
**Pull request by Innos**
Refactored code to use string.IsNullOrEmpty and interpolated strings and
hotfixed DOM strategy to work with IIFEs.

### CODE REVIEW

File: Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsPreprocessExecuteAndRunCodeAgainstUnitTestsWithMochaExecutionStrategy.cs [hunk 1]
```diff
@@ -171,11 +170,11 @@ protected virtual string BuildTests(IEnumerable<TestContext> tests, int numberOf
...
             var testsCode = string.Empty;
-            testsCode += @"
-                " + tests.FirstOrDefault().Input + @"
-                describe('JudgeTest1', function(){ 
-                " + UserInputPlaceholder + @"
-                });";
+            testsCode += $@"
+                {tests.FirstOrDefault().Input}
+                describe('JudgeTest1', function(){{ 
+                {UserInputPlaceholder}
```
Comment (KristianMariyanov) [inline Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsPreprocessExecuteAndRunCodeAgainstUnitTestsWithMochaExecutionStrategy.cs]: tab ->

File: Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsPreprocessExecuteAndRunJsDomUnitTestsExecutionStrategy.cs [hunk 2]
```diff
@@ -117,30 +112,32 @@ public class NodeJsPreprocessExecuteAndRunJsDomUnitTestsExecutionStrategy : Node
...
             foreach (var test in tests)
             {
-                var code = Regex.Replace(test.Input, "([\\\\\"'])", "\\$1");
-                code = Regex.Replace(code, "[\r\n\t]*", string.Empty);
+                var code = Regex.Replace(test.Input, "([\\\\`])", "\\$1");
 
                 testsCode += @"
 it('Test" + testsCount++ + @"', function(done) {
-    let content = '" + code + @"';
+    let content = `" + code + @"`;
```
Comment (KristianMariyanov) [inline Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsPreprocessExecuteAndRunJsDomUnitTestsExecutionStrategy.cs]: string interpolation?

File: Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsPreprocessExecuteAndRunJsDomUnitTestsExecutionStrategy.cs [hunk 3]
```diff
@@ -117,30 +112,32 @@ public class NodeJsPreprocessExecuteAndRunJsDomUnitTestsExecutionStrategy : Node
...
     let inputData = content.trim();
+    let code = {
+        run: " + UserInputPlaceholder + @"
```
Comment (KristianMariyanov) [inline Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsPreprocessExecuteAndRunJsDomUnitTestsExecutionStrategy.cs]: string interpolation?

File: Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsPreprocessExecuteAndRunJsDomUnitTestsExecutionStrategy.cs [hunk 4]
```diff
@@ -117,30 +112,32 @@ public class NodeJsPreprocessExecuteAndRunJsDomUnitTestsExecutionStrategy : Node
...
+    };
 
     let testFunc = new Function(" + this.TestFuncVariables + @", ""var result = this.valueOf();"" + inputData);
```
Comment (KristianMariyanov) [inline Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsPreprocessExecuteAndRunJsDomUnitTestsExecutionStrategy.cs]: string interpolation?

File: Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsPreprocessExecuteAndRunJsDomUnitTestsExecutionStrategy.cs [hunk 5]
```diff
@@ -117,30 +112,32 @@ public class NodeJsPreprocessExecuteAndRunJsDomUnitTestsExecutionStrategy : Node
...
-    testFunc.call(result, " + this.TestFuncVariables.Replace("'", string.Empty) + @");
+    testFunc.call(code.run, " + this.TestFuncVariables.Replace("'", string.Empty) + @");
```
Comment (KristianMariyanov) [inline Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsPreprocessExecuteAndRunJsDomUnitTestsExecutionStrategy.cs]: string interpolation?

File: Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsZipPreprocessExecuteAndRunUnitTestsWithDOMAndMochaExecutionStrategy.cs [hunk 6]
```diff
@@ -199,13 +198,12 @@ protected override string BuildTests(IEnumerable<TestContext> tests)
...
             {
-                var code = Regex.Replace(test.Input, "([\\\\\"'])", "\\$1");
-                code = Regex.Replace(code, "[\r\n\t]*", string.Empty);
+                var code = Regex.Replace(test.Input, "([\\\\`])", "\\$1");
 
                 testsCode += @"
                 it('Test" + testsCount++ + @"', function(done) {
                     this.timeout(10000);
-            	    let content = '" + code + @"';
+            	    let content = `" + code + @"`;
```
Comment (KristianMariyanov) [inline Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsZipPreprocessExecuteAndRunUnitTestsWithDOMAndMochaExecutionStrategy.cs]: string interpolation?

File: Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsPreprocessExecuteAndRunJsDomUnitTestsExecutionStrategy.cs (hunk 2 above)
Comment (KristianMariyanov) [inline Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsPreprocessExecuteAndRunJsDomUnitTestsExecutionStrategy.cs]: testCount and code

File: Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsPreprocessExecuteAndRunCodeAgainstUnitTestsWithMochaExecutionStrategy.cs (hunk 1 above)
Comment (Innos) [inline Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsPreprocessExecuteAndRunCodeAgainstUnitTestsWithMochaExecutionStrategy.cs]: I tabbed the {UserInputPlaceholder}!

**Pull request merged**